### PR TITLE
Add canonical string representation for identifiers

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
An identifier is represented as a string as a package identifier and a qualified name separated by a colon.

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- x ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala [hunk 1]
```diff
@@ -98,16 +104,14 @@ object Ref {
   }
 
   object DottedName {
-    type T = DottedName
```
Comment (remyhaemmerle-da) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala]: :+1:

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala [hunk 2]
```diff
@@ -142,13 +146,11 @@ object Ref {
     }
   }
 
-  case class QualifiedName private (module: ModuleName, name: DottedName) {
+  final case class QualifiedName private (module: ModuleName, name: DottedName) {
```
Comment (remyhaemmerle-da) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala]: :+1:

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala [hunk 3]
```diff
@@ -13,7 +13,7 @@ object Ref {
 
   /* Encoding of byte array */
   type HexString = IdString.HexString
-  val HexString = IdString.HexString
+  val HexString: HexStringModule[Ref.IdString.HexString] = IdString.HexString
```
Comment (remyhaemmerle-da) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala]:
```suggestion
  val HexString: IdString.HexString.type = IdString.HexString
```

Comment (stefanobaghino-da) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala]: Thanks, you're officially smarter than IntelliJ. :smile:

**Pull request merged**